content/news/2020_06_28_github_actions.md

```diff
@@ -0,0 +1,16 @@
+---
+title: "New GRASS GIS source code analysis with GitHub Actions"
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year ([here](/news/2019_05_17_svn_to_github_migration_started/) and [here](/news/2020_01_23_grass_gis_issues_now_on_github/)), a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis, in order to improve the overall quality, maintainability and standard compliance of our code.
+
+The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, on the other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```

Comment:
The OSGeo4W workflow is already committed and in place. I'm planning to copy out the built package as [a GitHub artifact](https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts) for daily builds, if needed, as we discussed in [#692](https://github.com/OSGeo/grass/pull/692). Having said that, I'm not sure if that's really necessary when we already have a working build process for daily OSGeo4W builds (where?).

Reply:
>I'm not sure if that's really necessary when we already have a working build process for daily OSGeo4W builds (where?).  

daily OSGeo4W builds are done at @landam server. 

in the long run, I suggest to use such things like GH action/GH arctifacts (or something else) to ease the burden from 1 maintainer of daily builds to a group.

AFAIK there are ideas to build/setup OSGeo4W itself in something similar like azure etc. thus, if GRASS is ready, the better.

@landam what do you think?

Reply:
If that's the case, +1 for daily artifacts.

Reply:
Updated accordingly